File: tests/integration-tests/src/main/java/com/jaynewstrom/json/sample/custom/LenientBooleanSerializer.java
package com.jaynewstrom.json.sample.custom;

import com.fasterxml.jackson.core.JsonGenerator;
import com.jaynewstrom.json.runtime.JsonSerializer;
import com.jaynewstrom.json.runtime.JsonSerializerFactory;

import java.io.IOException;

final class LenientBooleanSerializer implements JsonSerializer<Boolean> {
    @Override public void serialize(Boolean aBoolean, JsonGenerator jg, JsonSerializerFactory serializerFactory) throws IOException {
        jg.writeString(aBoolean.toString());
    }

    @Override public Class<?> modelClass() {
        return Boolean.class;
    }
}

File: tests/integration-tests/src/main/java/com/jaynewstrom/json/sample/custom/LenientBooleanSerializer.kt
package com.jaynewstrom.json.sample.custom

import com.fasterxml.jackson.core.JsonGenerator
import com.jaynewstrom.json.runtime.JsonSerializer
import com.jaynewstrom.json.runtime.JsonSerializerFactory

import java.io.IOException

internal object LenientBooleanSerializer : JsonSerializer<Boolean> {
    @Throws(IOException::class)
    override fun serialize(value: Boolean, jg: JsonGenerator, serializerFactory: JsonSerializerFactory) {
        jg.writeString(value.toString())
    }
}
